planner/core/rule_predicate_push_down.go

```diff
@@ -73,6 +79,8 @@ func splitSetGetVarFunc(filters []expression.Expression) ([]expression.Expressio
 
 // PredicatePushDown implements LogicalPlan PredicatePushDown interface.
 func (p *LogicalSelection) PredicatePushDown(predicates []expression.Expression) ([]expression.Expression, LogicalPlan) {
+	predicates = DeleteTrueExprs(p, predicates)
```

Comment:
It's a little redundant to remove it every time by checking the predicates passed down.
We could only do it for selection and join since they are generated the ones to be passed to their children

Reply:
en, we delete that in `func (p *LogicalUnionScan) PredicatePushDown`, but that in `func (ds *DataSource) PredicatePushDown(` is needed, because [PropagateConstant](https://github.com/pingcap/tidb/pull/24214/files#diff-9249fe4a6ce2585e14a72df590110c85414907dc8e9133045b2ddce0244acf45R111) may introduce some surely true expressions.